### PR TITLE
Basic consensus state store impl

### PR DESCRIPTION
**Pull request by askmish**
Signed-off-by: Ashish Kumar Mishra <ashish.k.mishra@intel.com>

### CODE REVIEW

File: src/core/src/database/lmdb.rs [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+ */
+
+use std::collections::HashMap;
+use std::path::Path;
+
+extern crate lmdb_zero as lmdb;
+
+use database::DatabaseError;
+
+const DEFAULT_SIZE: usize = 1 << 40; // 1024 ** 4
```
Comment (rranjan3) [inline src/core/src/database/lmdb.rs]: ** denotes _to the power_ here? Why not use a caret(^)?

File: src/core/src/database/mod.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+pub mod database;
+pub mod lmdb;
```
Comment (rranjan3) [inline src/core/src/database/mod.rs]: How would config module come into scope?

File: src/core/src/consensus_state_store.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+pub struct InMemoryConsensusStateStore {
+    consensus_state_by_block_id: HashMap<String, ConsensusState>,
+}
+
+impl InMemoryConsensusStateStore {
+    pub fn new() -> Self {
+        InMemoryConsensusStateStore::default()
+    }
+
+    fn get_consensus_state_by_block_id(&self, block_id: &str) -> Option<&ConsensusState> {
```
Comment (rranjan3) [inline src/core/src/consensus_state_store.rs]: _get_consensus_state_by_block_id_  block_id is but obvious here. _by_block_id_ not really needed in the function name.

File: src/core/src/database/lmdb.rs (hunk 1 above)
Comment (askmish) [reply src/core/src/database/lmdb.rs]: Both ** and ^ mean exponent

File: src/core/src/database/mod.rs (hunk 2 above)
Comment (askmish) [reply src/core/src/database/mod.rs]: We're not going to use the lmdb parts yet

**Pull request merged**